presto-hive/src/main/java/io/prestosql/plugin/hive/parquet/HdfsParquetDataSource.java

```diff
@@ -241,8 +241,15 @@ public static HdfsParquetDataSource buildHdfsParquetDataSource(
         DiskRange last = ranges.get(0);
         for (int i = 1; i < ranges.size(); i++) {
             DiskRange current = ranges.get(i);
-            DiskRange merged = last.span(current);
-            if (merged.getLength() <= maxReadSizeBytes && last.getEnd() + maxMergeDistanceBytes >= current.getOffset()) {
+            DiskRange merged = null;
+            boolean blockTooLong = false;
+            try {
+                merged = last.span(current);
+            }
+            catch (ArithmeticException e) {
```

Comment:
Wouldn't it be better to check for the problem condition instead of catching the exception afterwards?

Something like this:
```
            if (Math.max(last.getEnd(), current.getEnd() - Math.min(last.getOffset(), current.getOffset())) <= Integer.MAX_VALUE) {
                DiskRange merged = last.span(current);
                if (merged.getLength() <= maxReadSizeBytes && last.getEnd() + maxMergeDistanceBytes >= current.getOffset()) {
                    last = merged;
                }
            }
```

Reply:
The issue is that it duplicates the logic of DiskRange, so it's not much better. Alternatively, we could add a method to check it that would result in a valid span, but the current approach is good enough.